### PR TITLE
Parse buffer option

### PR DESCRIPTION
**Pull request by karimnaaji**
Fixes https://github.com/tangrams/tangram-es/issues/935

### CODE REVIEW

File: core/src/style/textStyleBuilder.cpp [hunk 1]
```diff
@@ -537,11 +537,13 @@ TextStyle::Parameters TextStyleBuilder::applyRule(const DrawRule& _rule,
             }
         }
 
-
         _rule.get(StyleParamKey::text_transition_hide_time, p.labelOptions.hideTransition.time);
         _rule.get(StyleParamKey::text_transition_selected_time, p.labelOptions.selectTransition.time);
         _rule.get(StyleParamKey::text_transition_show_time, p.labelOptions.showTransition.time);
 
+        if (!_rule.get(StyleParamKey::text_buffer, p.labelOptions.buffer)) {
```
Comment (tallytalwar) [inline core/src/style/textStyleBuilder.cpp]: Does the `buffer` value get inherited from parent `point` style if not specified? I think it is not in JS, which is what we have, but maybe good to confirm.

File: core/src/scene/styleParam.cpp [hunk 2]
```diff
@@ -180,6 +184,17 @@ StyleParam::Value StyleParam::parseString(StyleParamKey key, const std::string&
         }
         return vec.value;
     }
+    case StyleParamKey::text_buffer:
```
Comment (tallytalwar) [inline core/src/scene/styleParam.cpp]: Can `buffer` param take stops as a value? I could not find it in the documentation, but probably good to confirm from JS implementation.

File: core/src/style/textStyleBuilder.cpp (hunk 1 above)
Comment (karimnaaji) [inline core/src/style/textStyleBuilder.cpp]: It's unclear, have you tried and seen that there was no inheritance? I don't see any mention of that in https://github.com/tangrams/tangram/pull/306.

File: core/src/scene/styleParam.cpp (hunk 2 above)
Comment (karimnaaji) [inline core/src/scene/styleParam.cpp]: I don't think so, not mentioned in reference issue https://github.com/tangrams/tangram-docs/issues/141.

File: core/src/style/textStyleBuilder.cpp (hunk 1 above)
Comment (tallytalwar) [inline core/src/style/textStyleBuilder.cpp]: Yeah I couldn't find anywhere in the documentation for both the things I mentioned. Will have to give it a shot in play to confirm.

Comment (tallytalwar) [inline core/src/style/textStyleBuilder.cpp]: Yup doesn't look like this gets inherited, so we are good here.

File: core/src/scene/styleParam.cpp (hunk 2 above)
Comment (tallytalwar) [inline core/src/scene/styleParam.cpp]: Yup, stops seems to not work with JS either. So we are all good here.

**Pull request merged**